nkms/blockchain/eth/sol/source/contracts/MinersEscrow.sol

```diff
@@ -570,20 +519,18 @@ contract MinersEscrow is Issuer {
             return bytes32(info.value);
         } else if (_field == MinerInfoField.Decimals) {
             return bytes32(info.decimals);
-        } else if (_field == MinerInfoField.LockedValue) {
-            return bytes32(info.lockedValue);
-        } else if (_field == MinerInfoField.Release) {
-            return info.release ? bytes32(1) : bytes32(0);
-        } else if (_field == MinerInfoField.MaxReleasePeriods) {
-            return bytes32(info.maxReleasePeriods);
-        } else if (_field == MinerInfoField.ReleaseRate) {
-            return bytes32(info.releaseRate);
-        } else if (_field == MinerInfoField.ConfirmedPeriodsLength) {
-            return bytes32(info.confirmedPeriods.length);
-        } else if (_field == MinerInfoField.ConfirmedPeriod) {
-            return bytes32(info.confirmedPeriods[_index].period);
-        } else if (_field == MinerInfoField.ConfirmedPeriodLockedValue) {
-            return bytes32(info.confirmedPeriods[_index].lockedValue);
+        } else if (_field == MinerInfoField.StakesLength) {
+            return bytes32(info.stakes.length);
+        } else if (_field == MinerInfoField.StakeFirstPeriod) {
+            return bytes32(info.stakes[_index].firstPeriod);
+        } else if (_field == MinerInfoField.StakeLastPeriod) {
+            return bytes32(info.stakes[_index].lastPeriod);
+        } else if (_field == MinerInfoField.StakeLockedValue) {
+            return bytes32(info.stakes[_index].lockedValue);
+        } else if (_field == MinerInfoField.ConfirmedPeriod1) {
+            return bytes32(info.confirmedPeriod1);
+        } else if (_field == MinerInfoField.ConfirmedPeriod2) {
+            return bytes32(info.confirmedPeriod2);
```

Comment:
Btw could it be possible to return the whole thing with new version of solidity?

Reply:
I am trying to simplify it with changes from Byzantium fork